File: python/mirascope/llm/providers/model_id.py
from typing import TypeAlias

from .anthropic import (
    AnthropicModelId,
)
from .google import (
    GoogleModelId,
)
from .minimax import (
    MiniMaxModelId,
)
from .mlx import (
    MLXModelId,
)
from .openai import (
    OpenAIModelId,
)

ModelId: TypeAlias = (
    AnthropicModelId | GoogleModelId | OpenAIModelId | MLXModelId | MiniMaxModelId | str
)

File: typescript/src/llm/providers/model-id.ts
/**
 * Union of all provider model IDs.
 */

import type { AnthropicModelId } from '@/llm/providers/anthropic';
import type { GoogleModelId } from '@/llm/providers/google';
import type { OpenAIModelId } from '@/llm/providers/openai';

/**
 * Model identifier for any supported provider.
 *
 * This is a union of all provider-specific model IDs.
 */
export type ModelId =
  | AnthropicModelId
  | GoogleModelId
  | OpenAIModelId
  | (string & {});
